src/NBitcoin/ChainBase.cs

```diff
@@ -4,15 +4,33 @@
 
 namespace NBitcoin
 {
+    /// <summary>Base implementation for chains.</summary>
     public abstract class ChainBase
     {
+        /// <summary>
+        /// Gets the block given a block id(hash).
```

Comment:
i'd wrote ID as with capital I and D, I'd also put space between ID and `(`

but most importantly, here and in whole file, I think we should distinguish between Block and ChainedBlock
as Block is what represents and actual block, but ChainedBlock is imho misnamed, it is actually ChainedBlockHeader, or ChainedHeader, it is not a block 

So if we want to to keep this little confusing name ChainedBlock, we should at least make comments that make it clear that this is not a block. GetBlock is again misnamed method, it does not return a block

GetBlock should be named GetChainedBlock (optimally GetChainedBlockHeader ...) 

So I was wondering if we could at least start with comments being correct

Reply:
I would be up to renaming to `ChainedBlockHeader` even though ChainedHeader will also do the job (we don't have headers of anything else really).

Reply:
Yes I agree confusion between Blocks and Chained Block Headers. I will change all the comments for now and we can decide what to do about ChainedBlock, ChainBase, ConcurrentChain longer term.